src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -499,6 +507,23 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                                 {
                                     if (isRestoreSucceeded)
                                     {
+                                        foreach (RestoreSummary summary in restoreSummaries)
+                                        {
+                                            foreach (IRestoreLogMessage error in summary.Errors)
+                                            {
+                                                if (error.Code.Equals(NuGetLogCode.NU1901) || error.Code.Equals(NuGetLogCode.NU1902) || error.Code.Equals(NuGetLogCode.NU1903) || error.Code.Equals(NuGetLogCode.NU1904))
+                                                {
+                                                    await _infoBarService.Value.ShowInfoBar(t);
+                                                    break;
+                                                }
+                                            }
+                                        }
+
+                                        if (_infoBarService.IsValueCreated) // if the InfoBar was created and no vulnerabilities found, hide it.
+                                        {
```

Comment:
What is the scenario here? i.e. When do we expect this to happen?

Reply:
I'm glad you asked this. I originally thought, sure, it's how the infoBar is closed after a successful restore. 
But since it's immediately after `await _infoBarService.Value.ShowInfoBar(t);` , it reads like this just immediately hides the infoBar after showing it? 😕